chasm/tree.go

```diff
@@ -1433,3 +1512,60 @@ func serializeTask(
 
 	return blob, nil
 }
+
+// ExecutePureTask validates and then executes the given taskInstance against the
+// node's component. Executing an invalid task is a no-op (no error returned).
+func (n *Node) ExecutePureTask(baseCtx context.Context, taskInstance any) error {
+	registrableTask, ok := n.registry.taskFor(taskInstance)
+	if !ok {
+		return fmt.Errorf("unknown task type for task instance goType '%s'", reflect.TypeOf(taskInstance).Name())
+	}
+
+	if !registrableTask.isPureTask {
+		return fmt.Errorf("ExecutePureTask called on a SideEffect task '%s'", registrableTask.fqType())
+	}
+
+	// TODO - instantiate CHASM engine and attach to context
```

Comment:
we only need to do this for side effect tasks. Pure tasks should not call any chasm engine methods.